app/scripts/controllers/transactions/index.js

```diff
@@ -331,6 +331,195 @@ export default class TransactionController extends EventEmitter {
+    txId,
+    { estimatedBaseFee, decEstimatedBaseFee },
+  ) {
+    if (!this._validateTransaction(txId)) {
+      return;
+    }
+
+    const txEstimateBaseFees = {
+      estimatedBaseFee,
+      decEstimatedBaseFee,
+    };
+
+    const note = `Update Transaction Estimated Base Fees for ${txId}`;
+    this._updateTransaction(txId, txEstimateBaseFees, note);
+  }
+
+  /**
+   * updates a swap approval transaction with provided metadata and source token symbol
+   *  if the transaction state is unapproved.
+   *
+   * @param {string} txId
+   * @param {object} swapApprovalTransaction - holds the metadata and token symbol
+   * {
+   * type,
+   * sourceTokenSymbol
+   * }
+   * @param swapApprovalTransaction.type
+   * @param swapApprovalTransaction.sourceTokenSymbol
+   */
+  updateSwapApprovalTransaction(txId, { type, sourceTokenSymbol }) {
+    if (!this._validateTransaction(txId)) {
+      return;
+    }
+
+    const swapApprovalTransaction = { type, sourceTokenSymbol };
+    const note = `Update Swap Approval Transaction for ${txId}`;
+    this._updateTransaction(txId, swapApprovalTransaction, note);
+  }
+
+  /**
+   * updates a swap transaction with provided metadata and source token symbol
+   *  if the transaction state is unapproved.
+   *
+   * @param {string} txId
+   * @param {object} swapTransaction - holds the metadata
+   * {
+   * sourceTokenSymbol,
+   * destinationTokenSymbol,
+   * type,
+   * destinationTokenDecimals,
+   * destinationTokenAddress,
+   * swapMetaData,
+   * swapTokenValue,
+   *}
+   * @param swapTransaction.sourceTokenSymbol
+   * @param swapTransaction.destinationTokenSymbol
+   * @param swapTransaction.type
+   * @param swapTransaction.destinationTokenDecimals
+   * @param swapTransaction.destinationTokenAddress
+   * @param swapTransaction.swapMetaData
+   * @param swapTransaction.swapTokenValue
+   */
+  updateSwapTransaction(
+    txId,
+    {
+      sourceTokenSymbol,
+      destinationTokenSymbol,
+      type,
+      destinationTokenDecimals,
+      destinationTokenAddress,
+      swapMetaData,
+      swapTokenValue,
+    },
+  ) {
+    if (!this._validateTransaction(txId)) {
+      return;
+    }
+
+    const swapTransaction = {
+      sourceTokenSymbol,
+      destinationTokenSymbol,
+      type,
+      destinationTokenDecimals,
+      destinationTokenAddress,
+      swapMetaData,
+      swapTokenValue,
+    };
+    const note = `Update Swap Transaction for ${txId}`;
+    this._updateTransaction(txId, swapTransaction, note);
+  }
+
+  /**
+   * updates a transaction's user settings only if the transaction state is unapproved
+   *
+   * @param {string} txId
+   * @param {object} userSettings - holds the metadata
+   * { userEditedGasLimit, userFeeLevel }
+   * @param userSettings.userEditedGasLimit
+   * @param userSettings.userFeeLevel
+   */
+  updateTransactionUserSettings(txId, { userEditedGasLimit, userFeeLevel }) {
+    if (!this._validateTransaction(txId)) {
```

Comment:
We need to add another type of validation to all of these transactions. We do not want to allow any of these edits (in any of the methods added here) if the transaction status is not `unapproved`. In other words, `unapproved` in the only status a transaction can have if we are editing it with these methods. Otherwise we should return.

Reply:
Oh, yeah. I just forgot to rename the call to `_checkIfTxStatusIsUnapproved`